site/blog/2022-11-07-secure-rds-endpoints.mdx

```diff
@@ -0,0 +1,117 @@
+---
+slug: rds-privatelink
+title: Securely connect Amazon RDS using PrivateLink
+authors: [yrobla]
+---
+
+# Securely connect Amazon RDS using PrivateLink
+
+Do you have some database instances on RDS and wonder what's the most secure way to reach them?
+
+When creating a database, AWS is giving some specific information about the hostnames and ports to be used. You can access those to reach to your databases, by using specific clients for MySQL, Postgres, etc...:
+
+<img src="/img/how_to_connect_rds.png" />
+
+On a public VPC, those endpoints will be public by default. But you should consider creating them under a
+private VPC and access them internally, to grant additional security to those services that are critical.
+
+AWS PrivateLink offers the possibility to securely access those services without exposing them to internet,
+just using Amazon's private network.
+
+### Do you want to know if you have it properly configured?
+
+Please use this query to check it:
+
+<!--- https://www.urlencoder.org/ -->
+
+<a href="https://app.iasql.com/#/button/check_rds_endpoints/----%20Installing%20the%20needed%20modules%0ASELECT%20iasql_install%28%27aws_rds%27%29%3B%0ASELECT%20iasql_install%28%27aws_vpc%27%29%3B%0A%0A--%20Perform%20the%20query%20for%20endpoints%0Aselect%20RDS.region%2C%20vpc.is_default%2C%20vpc.cidr_block%2C%0A%28select%20count%28%2A%29%3E0%20from%20endpoint_interface%20where%20endpoint_interface.region%3DRDS.region%20and%20service%3D%27rds%27%20and%20endpoint_interface.vpc_id%3Dvpc.id%29%20as%20has_endpoint_interface%0Afrom%20RDS%20left%20outer%20join%20vpc%20on%20vpc.region%3DRDS.region%3B">
+  <button class="button button--lg button--primary">Run SQL to check your endpoints</button>
+</a>
+
+```
+ ---- Installing the needed modules
+SELECT
+  iasql_install ('aws_rds');
+
+SELECT
+  iasql_install ('aws_vpc');
```

Comment:
why not just one install statement? Also, isn't vpc automatically installed when rds is installed? I don't remember if it is a hard dependency for the module.